src/pages/on-chain-governance/[proposalId]/index.page.tsx

```diff
@@ -1,439 +1,145 @@
-                label="Payout address"
-                customStyle="lg:mb-[9px]"
-              />
-              <button
-                className="flex items-center bg-blue-100 lg:py-1.5 py-2.5 px-5 md:px-[17.5px] lg:px-[10.5px] rounded w-fit"
-                onClick={() => onClickCopy(proposal.payoutAddress ?? "")}
-                type="button"
-              >
-                <TextMiddleTruncate
-                  text={proposal.payoutAddress ?? ""}
-                  textLength={8}
-                  className="text-[#4A72DA] mr-3"
-                />
-                <CopyToClipboardIcon />
-              </button>
-            </div>
-          </>
-        )}
-      </div>
-    </div>
-  );
-}
-
-function DetailSectionTitle({
-  label,
-  customStyle,
-}: {
-  label: string;
-  customStyle?: string;
-}) {
-  return (
-    <span
-      className={classNames(
-        "inline-block text-gray-500 text-sm mb-2 lg:mb-3 tracking-[0.04px]",
-        customStyle
-      )}
-    >
-      {label}
-    </span>
-  );
-}
-
-function DetailSectionLink({ label, href }: { label: string; href: string }) {
-  return (
-    <a href={href} className="flex items-center">
-      <span className="text-lg text-[#4A72DA] mr-2 inline-block">{label}</span>
-      <IoMdOpen size={24} className="text-[#4A72DA]" />
-    </a>
-  );
-}
-
-function DetailSectionStatusTag({ type }: { type: ProposalStatus }) {
-  return (
-    <div
-      className={classNames(
-        {
-          "bg-green-100": type === ProposalStatus.COMPLETED,
-          "bg-red-100": type === ProposalStatus.REJECTED,
-        },
-        "py-2 px-3 rounded-[5px] flex items-center w-fit"
-      )}
-    >
-      <span
-        className={classNames(
-          {
-            "text-green-700": type === ProposalStatus.COMPLETED,
-            "text-red-700": type === ProposalStatus.REJECTED,
-          },
-          "text-sm font-medium lg:text-lg lg:font-semibold mr-2"
-        )}
-      >
-        {type === ProposalStatus.COMPLETED
-          ? "PROPOSAL ACCEPTED"
-          : "PROPOSAL REJECTED"}
-      </span>
-      {type === ProposalStatus.COMPLETED ? (
-        <CheckIcon />
-      ) : (
-        <RejectedIcon width={16} height={16} />
-      )}
-    </div>
-  );
-}
-
-function VotingProgressSection({
-  proposal,
-  currentBlockCount,
-  percYes,
-  percNo,
-  percNeutral,
-}: {
-  proposal: ProposalInfo;
-  currentBlockCount: number;
-  percYes: BigNumber;
-  percNo: BigNumber;
-  percNeutral: BigNumber;
-}) {
-  const { connection } = useNetwork();
-  let blockSeconds = 30;
-  switch (connection) {
-    case "Playground":
-      blockSeconds = 3;
-      break;
-    case "TestNet":
-      blockSeconds = 3;
-      break;
-    case "MainNet":
-    default:
-      blockSeconds = 30;
-  }
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [isChangeVoteClicked, setIsChangeVoteClicked] = useState(false);
+  const [isMasterNodeConfirmClicked, setIsMasterNodeConfirmClicked] =
```

Comment:
Why do we want to have these states are specifically for the voting dialog/pop up to be in the parent page? 
Some of them are not used in the page, e.g.: `isMasterNodeConfirmClicked`, `hasUserSelectedVote`, `hasUserSelectedVote`

Reply:
Check if we can replace all the flags with enums, by defining each voting sequence in states, e.g.: `Masternode input stage > Voting stage > Review stage > Generating stage > ready stage`